scripts/binary-install.js

```diff
@@ -64,23 +65,28 @@ class Binary {
 
     console.log(`Downloading release from ${this.url}`)
 
-    return axios({ ...fetchOptions, url: this.url, responseType: 'stream' })
-      .then(res => {
-        return new Promise((resolve, reject) => {
-          const sink = res.data.pipe(
-            tar.x({ strip: 1, C: this.installDirectory })
-          )
-          sink.on('finish', () => resolve())
-          sink.on('error', err => reject(err))
-        })
-      })
-      .then(() => {
-        // TODO: verify the blake32hash of the downloaded binary upon install
-        console.log(`${this.name} has been installed!`)
-      })
-      .catch(e => {
-        error(`Error fetching release: ${e.message}`)
+    try {
+      const response = await fetch(this.url, fetchOptions)
+
+      if (!response.ok) {
+        throw new Error(`Failed to download: ${response.status} ${response.statusText}`)
+      }
+
+      // Convert web stream to Node.js stream (CRITICAL FIX)
+      const nodeStream = Readable.fromWeb(response.body)
```

Comment:
This line in itself is fine, but I've seen a few Node.js bugs related to using this specific API (I mean that the Node.js logic was, at some point at least, buggy, for converting from web streams to native streams).